### ISSUE TITLE
Extend Gutenberg server rendering of templates and template parts for problem determination 

### ISSUE THREAD

**Issue by bobbingwide** [bobbingwide/fizzie#41]
While analysing the problems I came across with Gutenberg 9.4.0 I developed an extension to `template-part.php` which added debug information visible in the front end. 
It sandwiched the output for each template part in some debug `<div class="WP_DEBUG">` tags
which contained the name of the template part.

I then wrote some naff CSS to enable the information to be viewed on the front end. 

![image](https://user-images.githubusercontent.com/2474435/99889670-7a403980-2c4f-11eb-9e57-6ca1dc917c03.png)

### Requirements

- [x] Formalise this solution for template parts
- [x] Find a method to automatically produce front end information regarding the template file that's been selected
- [x] Consider ways to easily turn the debug information on and off. e.g. Additional constant ( FSE_DEBUG), logged in admin user.

**Comment by bobbingwide** [comment]
### Formalise solution for  template parts

In Fizzie's `template-part.php` I changed the code to:
```
/**
  * Produce some visual aid to the theme developer, showing where the template starts and ends.
  *
  */
   if ( defined( 'FSE_DEBUG') && FSE_DEBUG ) {
       if (defined('WP_DEBUG') && WP_DEBUG) {
           $wrapped = "<div class=\"WP_DEBUG\">template part start: $slug</div>";
           $wrapped .= $content;
           $wrapped .= "<div class=\"WP_DEBUG END\">template part end: $slug</div>";
           $content = $wrapped;
           global $_wp_current_template_content;
           bw_trace2($_wp_current_template_content, "current template content", false, BW_TRACE_DEBUG);
       }
   }
```
I didn't bother to check if the user was logged in, nor if they had theme editing capability.

### Find a method to produce information for the template file

- I couldn't see any easy way to extend Gutenberg code. 
- It'll have to be changed to fix the issue #39 - Gutenberg issue 27177
- So I'll just do something else for the time being

**Comment by bobbingwide** [comment]
Since I started creating the [SB-debug-block](https://github.com/bobbingwide/sb-debug-block) plugin, where I encountered a couple of problems, and since I've not done much work on the internationalization (i18n) of templates and template parts I haven't really put much thought into the debuggability of templates and template parts.

So I may as well close this issue. The last commit was delivered in v0.1.1.
